### ISSUE TITLE
Compression of a OME-TIFF file when writing

### ISSUE THREAD

**Issue by etiennebergeron** [filippocastelli/pyometiff#2]
Hello again,
I want to write OME-TIFF from big arrays (23000 X 23000 for example). Following the example in the README, this is how I generate an OME-TIFF with random values:
```
import pyometiff
import numpy as np

data = np.random.randn(1, 1, 3, 23000, 23000).astype('uint8')

metadata_dict = {
    "PhysicalSizeX" : "0.88",
    "PhysicalSizeXUnit" : "µm",
    "PhysicalSizeY" : "0.88",
    "PhysicalSizeYUnit" : "µm",
    "PhysicalSizeZ" : "3.3",
    "PhysicalSizeZUnit" : "µm",
    "Channels" : {
        "405" : {
            "Name" : "405nm",
            "SamplesPerPixel": 1,
            "ExcitationWavelength": 405.,
            "ExcitationWavelengthUnit": "nm"
        },
        "488" : {
            "Name" : "488nm",
            "SamplesPerPixel": 1,
            "ExcitationWavelength": 488.,
            "ExcitationWavelengthUnit": "nm"
        },
        "638" : {
            "Name" : "638nm",
            "SamplesPerPixel": 1,
            "ExcitationWavelength": 638.,
            "ExcitationWavelengthUnit": "nm"
        },
    }
}

writer = pyometiff.OMETIFFWriter("test_out.ome.tiff", dimension_order="ZTCYX", array=data, metadata=metadata_dict)
writer.write()
```
It works, but the output file is 1.47 GB big. Is there a way to compress it, in JPEG, for example?

Thanks for your help,

Etienne

**Comment by filippocastelli** [comment]
compression is handled by the underlying `tifffile` library dependency, which `pyometiff.OMETIFFWriter` uses in `omewriter.py` at line 82.

    def write_stack(self, array, xml_meta):
        self.use_bigtiff = self._use_bigtiff(array)
        self.use_bigtiff = True
        with tifffile.TiffWriter(str(self.fpath), bigtiff=self.use_bigtiff) as tif:
            tif.save(
                array, description=xml_meta, photometric=self.photometric, metadata=None, compression="deflate"
            )


Just by adding a `compression="deflate"` argument to the `TiffWriter.save` call we can see that your code produces a significantly lighter output file.

```
(pyometiff) phil@shelob:~/repos/pyometiff/scripts$ ls -lh
totale 372M
-rw-rw-r-- 1 phil phil 1003 lug 20 21:00 issue.py
-rw-rw-r-- 1 phil phil 372M lug 20 21:03 test_out.ome.tiff
```

I'll add a way for `pyometiff` to select compression schemes, thank you for opening the issue.

**Comment by filippocastelli** [comment]
`pyometiff.OMETIFFWriter` now (c824bfc0b185926e928f8d03d0c13787be62a701) has a `compression` argument that gets passed to `tifffile.TiffWriter.save` meaning that `tifffile`'s compression schemes are now supported in `pyometiff`.

You can find them at 
https://github.com/cgohlke/tifffile/blob/23ddc3256a3b55125a2c36e7cc14884dc3030027/tifffile/tifffile.py#L11603

Using compression in `tifffile` functionally adds an optional dependency on `imagecodecs`, which doesn't get installed by default with `tifffile` itself, I'll add it as a dependency in `pyometiff`.

I've noticed that the `"JPEG"` compression you suggested doesn't work particularly well with random data (the output file is even bigger than the original!), you might be better off with other compression schemes like `"DEFLATE"` (which in turn gets really good compression rates).

The resulting files are opened correctly by ImageJ as far as I can tell, that might not work for all compression schemes as it ultimately depends on the `tifffile` library.

I'll add this to the next `pyometiff` release, thank you again.

**Comment by filippocastelli** [comment]
Now included in v0.0.6 https://github.com/filippocastelli/pyometiff/tree/0.0.6

**Comment by etiennebergeron** [comment]
Perfect, thanks to you for this update! As the random data was mainly for tests, I will try the compression with a regular image and it should be fine. 

Cheers,

Etienne
